src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -396,6 +396,8 @@
     <!-- Non-null type is required for this node kind -->
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
 
+    <Field Name="IsDeclaration" Type="bool"/>
```

Comment:
We do have a separate node for d-declaration which is a statement. But all it contains is a `BoundDeconstructionAssignmentOperator`.
The problem is that in the case of an assignment that is not a declaration, the expressions on the LHS should make it into the node map. But in the case of a declaration, they should not.

For comparison, if you have `x = 1;`, you will have 3 nodes in the node map: one for the whole assignment expression, one for the value `1` and one for the local `x`. 
But if you have `var x = 1`, then you will have a different set of 3 nodes: one for the whole statement, one for the value `1` and one for the type `var`. But no `x`.